tests/functional/test_upload.py

```diff
@@ -275,6 +275,31 @@ def test_upload_for_fileobj(self):
         self.assert_expected_client_calls_were_correct()
         self.assert_upload_part_bodies_were_correct()
 
+    def test_limits_in_memory_chunks_for_fileobj(self):
+        # Limit the maximum in memory chunks to one but make number of
+        # threads more than one. This means that the upload will have to
+        # happen sequentially despite having many threads available because
+        # data is sequentially partitioned into chunks in memory and since
+        # there can only every be one in memory chunk, each upload part will
+        # have to happen one at a time.
```

Comment:
Yeah it is implicit to the ``self.add_*`` methods of the stubber. I will add comments to make that more clear about it implicitly having the assertion of order.